### PR TITLE
gitflows: add push to dockerhub

### PR DESCRIPTION
**Pull request by tynes**
## Description

Adds build and push to dockerhub for deployer docker image

### Fixes
- Fixes # [Link to Issue]

## Contributing Agreement

- [x] I have read and understood the [Optimism Contributing Guide and Code of Conduct](https://github.com/ethereum-optimism/optimism-monorepo/blob/master/.github/CONTRIBUTING.md) and am following those guidelines in this pull request.

### CODE REVIEW

File: .github/workflows/dockerhub-build-push.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        run: |
+          git clone https://github.com/ethereum-optimism/docker.git \
+              $HOME/docker
+          cd $HOME/docker
+          BRANCH=${GITHUB_REF##*/}
+          ./build.sh -s deployer -b $BRANCH
+          if [ $BRANCH == 'master' ]; then
+              docker push ethereumoptimism/deployer:latest
+          elif [[ $BRANCH == 'uat' || $BRANCH == 'testnet' ]]; then
+              docker push ethereumoptimism/deployer:$BRANCH
```
Comment (karlfloersch) [inline .github/workflows/dockerhub-build-push.yml:35]: Should we be pushing `conracts-v2`?

**Review by karlfloersch — APPROVED** [review]
This LGTM but I'm not 100% sure on the `docker push` logic -- is this pushing the wrong image?

File: .github/workflows/dockerhub-build-push.yml (hunk 1 above)
Comment (karlfloersch) [reply .github/workflows/dockerhub-build-push.yml:35]: Oh yeah my bad -- that makes sense. Totally forgot

**Comment by tynes** [comment]
This docker image is specifically for the `deployer` script

**Pull request merged**